src/compile/axis.ts

```diff
@@ -295,6 +326,41 @@ export namespace properties {
       }
     }
 
+    if (axis.tickLabelColor !== undefined) {
```

Comment:
@YuhanLu  

Looking closer, it seems like this method here will be throw error if `labelSpec` is `undefined`  (my old code starting from line 287 has the same problem too.  

Can you go through this method and make sure if it's still correct even though the input `labelSpec` is undefined? 

Also, make sure that this method never returns empty object (`{}`).